### PR TITLE
Modified GVAE forward, now outputs h instead of z

### PR DESCRIPTION
**Pull request by dnguyen1196**
In the previous version, `gvae.forward` would compute `z` vectors of each node (and then aggregate into individual subgraphs by summing the `z` vectorrs of the nodes in each subgraph). Where `z` is the mean of the approximate posterior distribution. It is the `z` that appears in the ELBO of the variational auto-encoder's loss function.

One potential problem with this approach was that our decoder uses `z` to predict link (bond) between atoms. The decoder would predict that there is a link between node `i` and `j` with probability `sigmoid(z_i.T @ z_j)`. 

This means that when we train the neural networks, we would "encourage" atoms/nodes that have bonds between them to have "similar" `z` vector representation. However, this is not true in chemistry because bonds can exist between very "different" atoms. In other words, we had an inductive bias that doesn't correspond directly to chemical properties of molecules.

Now the output from `gvae.forward` would be `h`, which is the output of the Graph convolution layers, and the input to the linear layer that outputs the mean and covariance of the approximate posterior distribution.

### CODE REVIEW

**Comment by karalets** [comment]
On it.
